objects/obj_main_menu/Step_0.gml

```diff
@@ -4,7 +4,7 @@
 main_menu_x += (main_menu_x_target - main_menu_x) / main_menu_speed;
 
 //Keyboard Controls
-if (main_menu_control)
+if (main_menu_control && !global.in_settings_menu)

```

Comment:
Does `in_settings_menu` need to be a global? Same with `player_has_control`? I feel like those values could just be set to true/false when needed and then we wouldn't need to have the overhead of a bunch of globals.

Reply:
I could 100% be misunderstanding globals, but to me they seem necessary if we're going to have something that isn't the settings menu trigger the settings menu appearing. Would absolutely love to discuss this in more detail, I could definitely be wrong about them needing to be global or there may be a trick I didn't think of to avoid that.

Reply:
used with() and/or obj.variable instead of globals